### ISSUE TITLE
Confusion about Property Optimization

### ISSUE THREAD

**Issue by Zhoushun2021** [jiaor17/DiffCSP#22]
Hello, I have a question in train a time-dependent energy prediction model. In the file energy_model.py, the loss_energy is about a property in the dataset, i.e. batch.y, for example it represents the formation energy. But why is the input to self.decoder batch.batch?

![image](https://github.com/user-attachments/assets/3c54ae47-40e8-4837-ac13-235c5416fde5)

**Comment by jiaor17** [comment]
Hi,
In Pytorch Geometric, `batch.batch` is a tensor that stores the batch indices for each node and helps keep track of which nodes belong to which original graph, you can see the source codes [here](https://pytorch-geometric.readthedocs.io/en/1.7.2/_modules/torch_geometric/data/batch.html) for more details. Usually, we use this tensor for scatter-based operations in GNNs, like a final pooling to aggregate graph features from node features.

**Comment by Zhoushun2021** [comment]
Thank you, does this loss_energy represent the loss of a property, such as formation energy. And does pred_e stand for the formation energy obtained from the prediction? Also, I have encountered the following problem during the training process, what should I do to solve  @jiaor17 
![image](https://github.com/user-attachments/assets/76dfe24b-a1bb-44fa-9169-183808dd2065)
